### PR TITLE
[MVS-537] Update zip code, phone, email, DOB validation logic and fix a spelling error

### PR DESCRIPTION
**Pull request by ShahdSawalhi**
## :floppy_disk: [Code Check]
_Make sure you've checked/done the following before making this PR!_
* Does your code build/run?
* Does your design approach make sense?
* Does your work achieve the intention of the card?
* Does your code have any unintentional side effects -- have you tested a full workflow & checked the logic branches you touched?
* Did you make a unit test for your code (as appropriate)?
* Is your branch relatively up to date with the Development branch (within ~2 commits)
* Have you moved the card to "Needs QA" in Jira

## :flipper: [JIRA Sprint Card]

https://mass-immunization-system.atlassian.net/browse/MVS-537

## :newspaper: [Work Description]

Fix the text fields validation logic. Make sure that when the user puts in an invalid input, it would proceed to the next page until it's fixed. This logic was updated on the phone number, email, zipcode, and DOB text fields in both the single and household registration workflows.
A spelling error found in the screening questions page in the PatReg app was also fixed in this card (includign vs including) 

## :vertical_traffic_light: [Testing/QA Notes]

- Run the PatReg app, make sure it compiles without any errors 
- Go through the single patient work flow 
- Try and input an invalid value for the zip code and click the button "continue". The user will not be able to proceed. Fix the input and click "continue" again the user will be able to move to the next page.
- Do the same thing with the phone number and email text fields in the Contact Info page, the DOB field in the personal info page, and the phone number field in the emergency contact page 
- When the Screening Questions page is reached, make sure there are no other spelling errors
- Repeat the same steps with the Household Registration workflow

### CODE REVIEW

File: PatientRegistrationApp/src/components/HouseholdContactInfo.vue [hunk 1]
```diff
@@ -106,8 +108,13 @@ import customerSettings from '../customerSettings'
 			phoneTypeOptions:['Home', 'Mobile', 'Work'],
 			emailRules: [
 				(v) =>
-				/^[\s]*$|.+@.+\..+/.test(v) ||
-				"Please provide a valid e-mail address",
+				/^[\s]*$|.+@.+\..+/.test(v) || 'Please provide a valid e-mail address',
+			],
+			phoneNumberRules: [
```
Comment (peterdobbs77) [inline PatientRegistrationApp/src/components/HouseholdContactInfo.vue:113]: <minor> for internationalization purposes, might want to refactor this to be `phoneNumberRulesUS`. See: https://en.wikipedia.org/wiki/North_American_Numbering_Plan </minor>

Comment (ShahdSawalhi) [reply PatientRegistrationApp/src/components/HouseholdContactInfo.vue]: I changed the variable name to phoneNumberRulesUs

**Comment by peterdobbs77** [comment]
Have you looked into https://vuelidate.js.org for validation?

**Comment by ShahdSawalhi** [comment]
> Have you looked into https://vuelidate.js.org for validation?

Yes I did. But in the PatReg app we already had a built in statement for our validation methods from previous cards. The scope of this card is to make sure that the user cannot advance to the next page if there is an invalid input. 

However I can change the scope of the card to use vuelidate if you think it is better. 
@dmcook5208 do you have any input on which method to use?

**Comment by peterdobbs77** [comment]
> 
> 
> > Have you looked into https://vuelidate.js.org for validation?
> 
> Yes I did. But in the PatReg app we already had a built in statement for our validation methods from previous cards. The scope of this card is to make sure that the user cannot advance to the next page if there is an invalid input.
> 
> However I can change the scope of the card to use vuelidate if you think it is better.
> @dmcook5208 do you have any input on which method to use?

Updating to vuelidate should probably be a separate refactor task. Increasing the scope of a task is usually not a good idea. 
I'd like to hear what Dana thinks, though.

**Review by peterdobbs77 — DISMISSED** [review]
Satisfied with your changes, but look into Vuelidate for future work/refactor. 
Merge conflict?

**Comment by david-fraley** [comment]
@ShahdSawalhi If we're going to look into changing how we validate things, that should be a separate card.

**Comment by dmcook5208** [comment]
> > Have you looked into https://vuelidate.js.org for validation?
> 
> Yes I did. But in the PatReg app we already had a built in statement for our validation methods from previous cards. The scope of this card is to make sure that the user cannot advance to the next page if there is an invalid input.
> 
> However I can change the scope of the card to use vuelidate if you think it is better.
> @dmcook5208 do you have any input on which method to use?

I haven't spent a lot of time digging into vee-validate and vuelidate (this is my first Vue project).  It would be good to more clearly understand the advantages of either of these libraries, but I don't think it's urgent since what we have is working and easy enough to follow.  I'm assuming vuelidate or vee-validate will allow us to achieve what is needed with less lines of code (which is a plus for longer-term maintainability).  Should we just create a card in the backlog for this so that we can come back to it when time permits?

**Comment by ShahdSawalhi** [comment]
Merge conflicts are resolved. 
Do you want me to add a card in the backlog for using vuelidate @dmcook5208 ?

**Review by dmcook5208 — APPROVED** [review]

**Pull request merged**